DuckDuckGo/Subscription/UserScripts/SubscriptionPagesUseSubscriptionFeature.swift

```diff
@@ -221,10 +235,29 @@ final class SubscriptionPagesUseSubscriptionFeature: Subfeature, ObservableObjec
         let accountManager = AccountManager()
         if let accessToken = accountManager.accessToken,
            case let .success(accountDetails) = await accountManager.fetchAccountDetails(with: accessToken) {
-            accountManager.storeAccount(token: accessToken, email: accountDetails.email, externalID: accountDetails.externalID)
-            emailActivationComplete = true
+            switch await SubscriptionService.getSubscriptionDetails(token: accessToken) {
+            
+            // If the account is not active, display an error and logout
+            case .success(let response) where !response.isSubscriptionActive:
+                transactionError = .failedToRestoreFromEmailSubscriptionInactive
+                accountManager.signOut()
+                return nil
```

Comment:
Do any of these functions marked as "throws" effectively throw any error ever?

Reply:
nop.  good call.  I'll just kill that.